packages/lit-dev-content/site/articles/article/redux-reactive-controllers.md

```diff
@@ -0,0 +1,633 @@
+while mixins are analogous to higher-order components.
+
+One advantage of the Reactive Controller pattern is that it creates a **with** relationship rather than an **is** relationship. For example, a component that uses a Reactive Controller that incorporates Redux logic is a **component with Redux selector abilities**, whereas a mixin that does the same would mean that the **component is a Redux selector component**. This type of composability results in code that is more portable, self-contained, and easier to refactor. This is because components that inherit via subclassing are more closely coupled with the logic they inherit.
+
+<img
+    src="/images/articles/redux-reactive-controllers/mixins-vs-controllers.webp"
+    loading="lazy"
+    title="Mixin Composition vs Reactive Controller Composition"
+    alt="two rectangle wire diagrams. The left is labeled 'Mixin composition' the right 'Reactive Controller Composition'. There is a key that says blue lines are instance properties and green lines are class / superclasses. The mixin composition is a green rectangle labeled 'SideDrawerElement' with a co-centric green rectangle inside of it labeled 'AsyncTaskElement()`, with another co-centric green rectangle labeled 'ReduxSelectorElement()', with a final co-centric green rectangle labeled 'LitElement'. The Reactive controller diagram is a green rectable labeled 'SideDrawerElement' with three rectangles inside of it. They are: a blue dashed rectangle labeled 'ReduxSelectorController' and another one labeled 'AsyncTaskController'. The third rectangle is a green rectangle labeled 'LitElement'.">
+
+Reactive Controllers are just an interface – a pattern, which makes them easier to use with other component systems without committing to a specific architecture. This makes it possible to create Reactive Controller adapters that work with other frameworks and libraries, such as [React](https://www.npmjs.com/package/@lit/react#usecontroller), Vue, Angular, and Svelte.
+
+## What is Redux?
+
+[Redux](https://redux.js.org/) is a mature library that introduces patterns to manage state across a JavaScript application. The Lit team does not have a particular endorsement for a single state management library, but we will be using Redux as an example for creating a Reactive Controller, because the patterns used in integrating Redux into Lit with a Reactive Controller may be used to integrate for other popular libraries.
+
+There are three general concepts that Redux describes:
+
+1. Stores
+2. Actions
+3. Reducers
+
+Stores contain the current application state. Actions describe what kind of change to make to the state, and reducers take actions and apply them to the current state to return a new state. Here is a diagram derived from the [official Redux documentation](redux.js.org/tutorials/essentials/part-1-overview-concepts#redux-application-data-flow ?) that depicts the interaction pattern between these concepts:
```

Comment:
This markdown link doesn't work, and seems to render in full.

See preview url: https://pr1283-b46091e---lit-dev-5ftespv5na-uc.a.run.app/articles/redux-reactive-controllers/#:~:text=%5Bofficial%20Redux%20documentation